Client/Experiments/Messaging/GleanPlumbMessageStore.swift

```diff
@@ -51,6 +51,12 @@ class GleanPlumbMessageStore: GleanPlumbMessageStoreProtocol {
     func onMessageDisplayed(_ message: GleanPlumbMessage) {
         message.metadata.impressions += 1
 
+        if message.metadata.impressions >= message.style.maxDisplayCount {
+            onMessageExpired(message.metadata,
+                             surface: message.data.surface,
+                             shouldReport: false)
+        }
```

Comment:
This is _almost_ there!

```suggestion
        if message.isExpired {
            onMessageExpired(message.metadata,
                             surface: message.data.surface,
                             shouldReport: true)
        }
```